src/gui/dragAndDrop.js

```diff
@@ -2,48 +2,197 @@
  * Handles processing of drag and drop events for the mathematical Nodes
  */
 
+// Global imports
 /* global document */
 /* global NodeForest */
 /* global Calculator */
 /* global Operators */
 /* global NodeFactory */
+/* global NumberNode */
 
+/**
+ * Sets data attribute to hold the id of the picked up HTML Node
+ * @param event
+ */
 function pickUpNode(event) {
     event.dataTransfer.setData("text", event.target.id);
-    event.dataTransfer.setData("parent", event.target.parentNode.id);
 }
 
+function handlePanelNode(node, target, event) {
+    const clone = node.cloneNode(true);
+    clone.id = NodeFactory.createNode(clone.classList, false);
+    const backendNode = NodeForest.getNode(clone.id);
+
+    // Set the number to 0 on initial drop if number node
+    if (node.classList.contains("number")) {
+        NodeForest.getNode(clone.id).number = 0;
+    }
+
+    // Insert into a binary node
+    if (target.id.toString().endsWith("Left")) {
+        NodeForest.getNode(target.parentNode.id).setLeftOperand(backendNode);
+    }
+    else if (target.id.toString().endsWith("Right")) {
+        NodeForest.getNode(target.parentNode.id).setRightOperand(backendNode);
+    }
+    target.appendChild(clone);
+}
+
+function handlePanelEquationPane(node, target, event) {
+    const clone = node.cloneNode(true);
+    clone.id = NodeFactory.createNode(clone.classList, true);
+
+    target.appendChild(clone);
+    clone.style.position = "fixed";
+    clone.style.top = `${event.pageY}px`;
+    clone.style.left = `${event.pageX}px`;
+}
+
+function handleRootNodeNode(node, target, event) {
```

Comment:
Is this name intentional?

Reply:
"NodeNode"

Reply:
This is intentional. "Handle root nodes when dragged into nodes"